drivers/gpu/drm/vc4/vc4_firmware_kms.c

```diff
@@ -1097,6 +1098,17 @@ vc4_crtc_mode_valid(struct drm_crtc *crtc, const struct drm_display_mode *mode)
 		break;
 	}
 
+	/* Pi4 can't generate odd horizontal timings on HDMI, so reject modes
+	 * that would set them.
+	 */
+	if (fkms->bcm2711 &&
+	    (vc4_crtc->display_number == 2 || vc4_crtc->display_number == 7) &&
+	    ((mode->hdisplay |				/* active */
+	     (mode->hsync_start - mode->hdisplay) |	/* front porch */
```

Comment:
Sorry to be a stickler, but this line and the following few would benefit from an extra preceding space - everything else is so beautifully indented.

Reply:
You can go off a person you know!

Reply:
Only if you like them in the first place.